src/Orchard.Web/Modules/Orchard.Fields/Drivers/BooleanFieldDriver.cs

```diff
@@ -35,7 +35,7 @@ public class BooleanFieldDriver : ContentFieldDriver<BooleanField> {
 
         protected override DriverResult Editor(ContentPart part, BooleanField field, dynamic shapeHelper) {
             // if the content item is new, assign the default value
-            if(!part.HasDraft() && !part.HasPublished()) {
+            if(!field.Value.HasValue) {
```

Comment:
@TFleury 

Working on field drivers, i noticed that with the boolean field you can use a neutral option in settings, e.g `Maybe`. So, if you select `Maybe`, then `!field.Value.HasValue` will be true without meaning it's a new content item. This because here null is considered as a valid value.

If, for another reason, you still need to change this test, i think we can find the right one.

Best